File: core4/leetcode/Arrays/P27/main.py
class Solution:
    
    def RemoveElement(self, nums, val):
        if not nums:
            return 0
            
        k = 0
        
        for i in range(len(nums)):
            if nums[i] != val:
                nums[k] = nums[i]
                k += 1 
        
        return k


if __name__ == "__main__":
    nums = [0, 1, 2, 2, 3, 0, 4, 2]
    val = 2  # value to remove
    
    obj = Solution()
    K = obj.RemoveElement(nums, val)
    
    print(f"Number of elements after removing {val}: {K}")
    print("New array:", nums[:K])

File: core4/leetcode/Arrays/P27/main.cpp
#include <iostream>
#include <vector>
#include <unordered_map>
using namespace std;


#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int RemoveElement(vector<int>& nums, int val) {
        if (nums.empty()) return 0;

        int k = 0;
        for (int i = 0; i < nums.size(); i++) {
            if (nums[i] != val) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
};

int main() {
    Solution object;
    vector<int> nums = {0,1,2,2,3,0,4,2};
    int val = 2;

    int k = object.RemoveElement(nums, val);
    cout << "K is: " << k << endl;

    cout << "New nums: ";
    for (int i = 0; i < k; i++) {
        cout << nums[i] << " ";
    }
    cout << endl;

    return 0;
}














